### PR TITLE
Fixes #1381 Forcing App in portrait mode.

### LINKED ISSUE #1381: Overflow on set_url page in landscape mode
**Describe the bug**
when we rotate the screen in landscape-mode on set_url page their is a overflow of 110 pixels in my device. 
also we cannot set url in landscape mode.
**To Reproduce**

1. open the app
2. go to set url page
3. rotate the phone by 90 degrees 

**Expected behavior**
It should have not overflowed and alowwed to set url in landscape mode

**Actual behavior**
It overflowed

**Screenshots**
issue
issue2

### PR DESCRIPTION
**Pull request by Ayush0Chaudhary**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #1381

**Did you add tests for your changes?**
no

**Snapshots/Videos:**

https://user-images.githubusercontent.com/95746190/210848636-a6010bc3-9fe1-4db0-bd62-7095836eba9b.mp4

**If relevant, did you update the documentation?**

**Summary**

fixed the overflow of the qr in landscape mode. I used **LayoutBuilder** to manage the constraints

**Does this PR introduce a breaking change?**
NO

**Other information**
NO

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**
yes

### CODE REVIEW

**Comment by palisadoes** [comment]
@Ayush0Chaudhary Please update the test cases for this file based on your updates. We need to keep increasing our test code coverage.

**Review by SiddheshKukade — DISMISSED** [review]
Changes are looking fine to be merged. We also need to update the tests for this file.

**Review by palisadoes — CHANGES_REQUESTED** [review]
@Ayush0Chaudhary Please update the test cases for this file based on your updates. We need to keep increasing our test code coverage.

**Comment by Ayush0Chaudhary** [comment]
Adding the changes asap!

**Comment by Ayush0Chaudhary** [comment]
Screenshot from 2023-01-07 15-18-30


I am not able to add test for portrait mode, **can you help or direct me to some resource I can refer to**

Couldn't find proper method on google too.
@palisadoes @SiddheshKukade

**Comment by Ayush0Chaudhary** [comment]
orientation setter is only used by me in whole app
Screenshot from 2023-01-08 16-42-12

I read 8-9 tests
But couldn't find orientation setter, the default size of screen in unittest is
Size(**2400.0, 1800.0**)
thus it is testing for **landscape but not portrait**


@palisadoes @SiddheshKukade can you help with this??

**Comment by palisadoes** [comment]
@Ayush0Chaudhary We have decided that it will be best to disable landscape completely as it will make the app more difficult to use. 

However we should think of ways to have a standard error when overflows like this occur. Please also work with @TheHazeEffect as he says he knows of a potential solution. This will be a good addition to the code base.

There is another PR https://github.com/PalisadoesFoundation/talawa/pull/1380 where we are working on a widget to create standardized error messages. Coordinating with that PR would be good too so that you don't duplicate work.

**Comment by Ayush0Chaudhary** [comment]
There won't be duplicate work, as I am also the author of #1380 @palisadoes. I will disable the landscape in my next Commit ASAP! 
@TheHazeEffect Can you suggest how can we go after this issue??

**Comment by Ayush0Chaudhary** [comment]
@palisadoes can you merge the PR now??

**Comment by palisadoes** [comment]
@TheHazeEffect @SiddheshKukade please review

File: lib/main.dart [hunk 1]
```diff
@@ -140,6 +141,12 @@ class _MyAppState extends State<MyApp> {
     initQuickActions();
     FetchPluginList();
     super.initState();
+    fs.SystemChrome.setPreferredOrientations(
+      [
+        fs.DeviceOrientation.portraitUp,
+        fs.DeviceOrientation.portraitDown,
+      ],
+    );
```
Comment (SiddheshKukade) [inline lib/main.dart:149]: This looks OK to lock the app in portrait mode. 👍🏻

**Review by SiddheshKukade — COMMENTED** [review]
I'm not sure why you've deleted the `set_url_view_model_test.dart` file. I think It was just a test file to check if the QR works.

Also, I will suggest renaming the PR to something like "Forcing Portrait Mode on the app"

**Comment by Ayush0Chaudhary** [comment]
> I'm not sure why you've deleted the `set_url_view_model_test.dart` file. I think It was just a test file to check if the QR works.
> 
> Also, I will suggest renaming the PR to something like "Forcing Portrait Mode on the app"

I made the changes, @SiddheshKukade , I am new to GitHub so apologies.

**Comment by palisadoes** [comment]
@Ayush0Chaudhary  Our code coverage levels have decreased with this PR. Please add tests that cover the changes you made.

image

**Review by palisadoes — CHANGES_REQUESTED** [review]
@Ayush0Chaudhary Please update the test cases for this file based on your updates. We need to keep increasing our test code coverage.

image

**Comment by palisadoes** [comment]
@Ayush0Chaudhary Please also fix the failing tests

**Comment by Ayush0Chaudhary** [comment]
The Test is failing because of some issue in HiveDb https://github.com/hivedb/hive/issues/942 (some other people with similar issue)

The failing test is **app_setting_page_test**

------------

If I **just** run the Above test, it works fine
Screenshot from 2023-01-11 08-43-27

But when I run it with **flutter test --coverage**, this test fail
Screenshot from 2023-01-11 08-44-08

--------------
**The Error shown**
```

00:04 +6 -1: loading /home/panda/talawa/test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart [E]                                                                                                        
  Failed to load "/home/panda/talawa/test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart": lock failed, path = 'temporaryPath/currentuser.lock' (OS Error: Resource temporarily unavailable, errno = 11)
  package:hive/src/backend/vm/storage_backend_vm.dart 81:5                           StorageBackendVm.initialize
  ===== asynchronous gap ===========================
  package:hive/src/hive_impl.dart 123:9                                              HiveImpl._openBox
  ===== asynchronous gap ===========================
  package:hive/src/hive_impl.dart 154:12                                             HiveImpl.openBox
  ===== asynchronous gap ===========================
  test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart 98:3  main
  ===== asynchronous gap ===========================
  /tmp/flutter_tools.XKEDKN/flutter_test_listener.FFEHYH/listener.dart 19:3          _testMain

```


Its not related to the my work @palisadoes @SiddheshKukade

**Comment by Ayush0Chaudhary** [comment]
@palisadoes @SiddheshKukade

**Comment by palisadoes** [comment]
@TheHazeEffect Your input would we welcome here too.

**Comment by SiddheshKukade** [comment]
Hi, @Ayush0Chaudhary maybe you've missed `deleteFromDisk()` somewhere. Even if it's falling on your side it should work fine here on CI/CD if the code is okay. But as the pipeline here is also failing that looks like the problem is in your code. please check again.

**Comment by palisadoes** [comment]
> Hi, @Ayush0Chaudhary maybe you've missed `deleteFromDisk()` somewhere. Even if it's falling on your side it should work fine here on CI/CD if the code is okay. But as the pipeline here is also failing that looks like the problem is in your code. please check again.

1. I agree Siddesh, if you look at the history of our GitHub actions the failures only occur on this PR. https://github.com/PalisadoesFoundation/talawa/actions 
2. There is one PR workflow that fails that I was attempting, but I closed the PR, so it is not the cause.

@Ayush0Chaudhary Please take a closer look.

**Comment by Ayush0Chaudhary** [comment]
@palisadoes @SiddheshKukade can we merge now??? All test are passed!!

Screenshot from 2023-01-20 02-46-55

**Comment by SiddheshKukade** [comment]
Please update the code as the tests here are failing.

**Comment by Ayush0Chaudhary** [comment]
Screenshot from 2023-01-26 17-05-54

**Comment by Ayush0Chaudhary** [comment]
Screenshot from 2023-01-26 17-07-11

**Comment by Ayush0Chaudhary** [comment]
@SiddheshKukade please check, code coverage is 68.3% now.

**Comment by Ayush0Chaudhary** [comment]
Ok got it

**Comment by Ayush0Chaudhary** [comment]
@noman2002 Can we make this file unit-test as different issue?? I am facing difficulties, I cannot examples for main.dart test anywhere??

**Comment by noman2002** [comment]
> @noman2002 Can we make this file unit-test as different issue?? I am facing difficulties, I cannot examples for main.dart test anywhere??

@palisadoes what you say ?

**Comment by palisadoes** [comment]
Yes, please create a separate issue for the tests

**Comment by Ayush0Chaudhary** [comment]
Ok I am creating the issue to write test of main.dart

**Comment by Ayush0Chaudhary** [comment]
So can we merge this PR ??@noman2002

File: test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart [hunk 2]
```diff
@@ -85,19 +80,10 @@ Widget createChangePassScreenDark({ThemeMode themeMode = ThemeMode.dark}) =>
     );
 
 Future<void> main() async {
-  TestWidgetsFlutterBinding.ensureInitialized();
   testSetupLocator();
+  TestWidgetsFlutterBinding.ensureInitialized();
   locator<GraphqlConfig>().test();
   locator<SizeConfig>().test();
```
Comment (noman2002) [inline test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart:86]: Please restore this file. We don't need to change this file.

**Review by noman2002 — CHANGES_REQUESTED** [review]

File: test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart (hunk 2 above)
Comment (noman2002) [reply test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart:86]: Its the same as previous one @Ayush0Chaudhary

Comment (Ayush0Chaudhary) [reply test/widget_tests/after_auth_screens/app_settings/app_setting_page_test.dart:91]: replaced it.

**Comment by Ayush0Chaudhary** [comment]
@noman2002 made the changes you asked for

**Review by noman2002 — CHANGES_REQUESTED** [review]
Please resolve the conflicts.

**Review by noman2002 — APPROVED** [review]

**Comment by Ayush0Chaudhary** [comment]
@noman2002 please review.
@palisadoes

**Comment by Ayush0Chaudhary** [comment]
@palisadoes @SiddheshKukade @TheHazeEffect please approve the changes!!

**Comment by Ayush0Chaudhary** [comment]
@palisadoes is there a problem you see with PR, I can change it if you want

**Comment by noman2002** [comment]
> @palisadoes is there a problem you see with PR, I can change it if you want

There is a huge time difference between India and Jamaica. Wait for him to review the changes.

**Comment by Ayush0Chaudhary** [comment]
> > @palisadoes is there a problem you see with PR, I can change it if you want
> 
> There is a huge time difference between India and Jamaica. Wait for him to review the changes.

ok @noman2002, thank you

**Review by palisadoes — APPROVED** [review]

**Pull request merged**